Automattic/jetpack
# Protect: Add IP block and allow list toggles

### Description
Updates the Protect Firewall UI to allow the IP block and allow lists to be toggled independently.

## Proposed changes:

* Makes use of the new `jetpack_waf_ip_allow_list_enabled` / `jetpack_waf_ip_block_list_enabled` settings, replacing `jetpack_waf_ip_list`
* Removes the dedicated page manual rules and adds independent toggles for each IP list to the primary Firewall page
* Modifies sections dependent on the original setting

### Other information:


## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1262 

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Start up JT, and enable Protect
* Proceed to Firewall settings and toggle the various options in different combinations
* Ensure the settings all perform as expected and that no regressions are introduced
* Manually set `Waf_Runner::is_unsupported_environment` to false and verify that the BFP allow list still functions
* Upgrade and downgrade and ensure all expected settings are still accessible and functional
* Verify that a blocklisted IP cannot make requests or access the site admin when BFP is enabled
* Ensure that the blocked IP can once again do so once whitelisted
* Ensure cross compatibility exists when Jetpack is enabled and the settings are toggled from that end
* Ensure backward compatibility exists when the update is applied to an existing site with the prior settings in place

## Screenshots

Enabled:
![Screen Shot 2024-07-17 at 08 55 18](https://github.com/user-attachments/assets/68e73596-247e-4401-95d0-7d5bad14fb13)

Disabled:
![Screen Shot 2024-07-17 at 08 57 01](https://github.com/user-attachments/assets/d8d709d0-c39c-4a4b-b51b-eceda504e9df)

WAF Disabled:
![Screen Shot 2024-07-17 at 08 58 07](https://github.com/user-attachments/assets/9c778b34-b766-41d9-8b26-28722da11d99)

Unsupported Env:
![Screen Shot 2024-07-17 at 09 04 33](https://github.com/user-attachments/assets/8789b213-2f12-449c-acb1-86f9acdd04d9)
